carp.deployments.core/src/commonMain/kotlin/dk/cachet/carp/deployments/infrastructure/versioning/DeploymentServiceApiMigrator.kt

```diff
@@ -48,6 +47,35 @@ private val major1Minor1To3Migration =
                     }
                     ApiResponse( responseObject, response.ex )
                 }
+                "dk.cachet.carp.deployments.infrastructure.DeploymentServiceRequest.CreateStudyDeployment",
+                "dk.cachet.carp.deployments.infrastructure.DeploymentServiceRequest.GetStudyDeploymentStatus",
+                "dk.cachet.carp.deployments.infrastructure.DeploymentServiceRequest.UnregisterDevice",
+                "dk.cachet.carp.deployments.infrastructure.DeploymentServiceRequest.DeployDevice",
```

Comment:
We only have coverage for `CreateStudyDeployment`, if I remember correctly that was the only failing test, as [ Actual testing of the status responses should already be covered adequately in StudyDeployment tests.](https://github.com/cph-cachet/carp.core-kotlin/blob/07dbd91377353990db18f1bc50c6efd6665bae60/carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/DeploymentServiceTest.kt#L101), but the regression test case are generated from `DeploymentServiceTest`.

> Extra test cases seem relevant

Do we only want the generated json request files, or we also want those tests in `DeploymentServiceTest`? I think I'll include them first anyways to agree on the test cases, and they can be removed later if not desired. 

I added it to `1.0` because I felt the test case was needed from the start of this major version, not due to some changes introduced in `1.1`. Then simlly forget to add them `1.1` 😛. As I generated those json files in ver. 1.0 and see a list of diff with origional ones, wandering what I breaked and realized those are the generated UUIDs difference...

Reply:
> Do we only want the generated json request files, or we also want those tests in DeploymentServiceTest?

I'd always keep them in sync, so that it's just a matter of copying the output of all of them over. 

But, obviously the comment I left in code there didn't consider the implications on reduced regression testing of the infrastructure layer (JSON) by not covering all potential input/output. 😅 So albeit valid—there is plenty of functional coverage in StudDeploymentTest—, that doesn't cover serialization and JSON schema definitions.

I think it would make sense to write single test cases that try to cover as much as possible, rather than writing them as unit tests. These are more or less integration tests after all.